docs/book/15-Exceptions.md

```diff
@@ -48,7 +48,7 @@ if(t == null)
 
 这就抛出了异常，于是在当前环境下就不必再为这个问题操心了，它将在别的地方得到处理。具体是哪个“地方”后面很快就会介绍。
 
-异常使得我们可以将每件事都当作一个事务来考虑，而异常可以看护着这些事务的底线“…事务的基本保障是我们所需的在分布式计算中的异常处理。事务是计算机中的合同法，如果出了什么问题，我们只需要放弃整个计算。”我们还可以将异常看作是一种内建的恢复（undo）系统，因为（在细心使用的情况下）我们在程序中可以拥有各种不同的恢复点。如果程序的某部分失败了，异常将“恢复”到程序中某个已知的稳定点上。
+异常允许你将做的每件事都当作一个事务来考虑，而异常守护着这些事务：“......事务的基本保障是我们所需的分布式计算中的异常处理。事务相当于计算机中的合同法。如果任何事出现了错误，我们只需要扔掉整个计算。”你也可以将异常看作一种内建的“恢复”（undo）系统，因为（在细心使用时）你在程序中可以有各种恢复点。如果程序的一个部分失败了，异常将“恢复”到一个已知的稳定点上。
```

Comment:
改了之后很奇怪。贴原文

Reply:
Exceptions allow you to think of everything you do as a transaction, and the exceptions guard those transactions: "...the fundamental premise of transactions is that we needed exception handling in distributed computations. Transactions are the computer equivalent of contract law. If anything goes wrong, we'll just blow away the whole computation." You can also think about exceptions as a built-in “undo” system, because (with some care) you can have various recovery points in your program. If a part of the program fails, the exception will “undo” back to a known stable point in the program.

Reply:
the fundamental premise of transactions is that we needed exception handling in distributed computations.

这个是我们需要分布式计算的异常处理机制。

不应该翻译成「我们所需的分布式计算中的异常处理」 赶紧很怪

Reply:
「we'll just blow away the whole computation.」
不翻译成扔掉吧，赶紧太白话了。

改成「丢弃」吧

Reply:
如果程序的一个部分失败了

这个 if 翻译成 「一旦」吧

Reply:
> the fundamental premise of transactions is that we needed exception handling in distributed computations.
> 
> 这个是我们需要分布式计算的异常处理机制。
> 
> 不应该翻译成「我们所需的分布式计算中的异常处理」 赶紧很怪

如果该成这样：“事务的基本保障是我们需要分布式计算的异常处理机制。”是个病句，事务的基本保障是我们的一种需要。

Reply:
> 如果程序的一个部分失败了
> 
> 这个 if 翻译成 「一旦」吧

如果和一旦相差不大，你觉得需要改就直接改。

Reply:
> > 如果程序的一个部分失败了
> > 这个 if 翻译成 「一旦」吧
> 
> 如果和一旦相差不大，你觉得需要改就直接改。

含义差距不大，但是从阅读的体验上面来看 「一旦」更好

Reply:
> > the fundamental premise of transactions is that we needed exception handling in distributed computations.
> > 这个是我们需要分布式计算的异常处理机制。
> > 不应该翻译成「我们所需的分布式计算中的异常处理」 赶紧很怪
> 
> 如果该成这样：“事务的基本保障是我们需要分布式计算的异常处理机制。”是个病句，事务的基本保障是我们的一种需要。

为啥就变成了一个病句了。

事务的基本保障是，我们需要分布式计算的异常处理机制。

保障是我们需要xxx的机制

或者难受就改变语序

我们需要的机制是保障

Reply:
> > > 如果程序的一个部分失败了
> > > 这个 if 翻译成 「一旦」吧
> > 
> > 
> > 如果和一旦相差不大，你觉得需要改就直接改。
> 
> 含义差距不大，但是从阅读的体验上面来看 「一旦」更好

这本书错翻，暴力直译，不知所云的地方太多了。这点差别就是小巫见大巫。。。

Reply:
> > > the fundamental premise of transactions is that we needed exception handling in distributed computations.
> > > 这个是我们需要分布式计算的异常处理机制。
> > > 不应该翻译成「我们所需的分布式计算中的异常处理」 赶紧很怪
> > 
> > 
> > 如果该成这样：“事务的基本保障是我们需要分布式计算的异常处理机制。”是个病句，事务的基本保障是我们的一种需要。
> 
> 为啥就变成了一个病句了。
> 
> 事务的基本保障是，我们需要分布式计算的异常处理机制。
> 
> 保障是我们需要xxx的机制
> 
> 或者难受就改变语序
> 
> 我们需要的机制是保障

“事务的保障是我们需要xxx的机制”事务的保障变成了我们需要。事务的保障应该是一个名词，而不是一个动词或者动作。

Reply:
> > > > the fundamental premise of transactions is that we needed exception handling in distributed computations.
> > > > 这个是我们需要分布式计算的异常处理机制。
> > > > 不应该翻译成「我们所需的分布式计算中的异常处理」 赶紧很怪
> > > 
> > > 
> > > 如果该成这样：“事务的基本保障是我们需要分布式计算的异常处理机制。”是个病句，事务的基本保障是我们的一种需要。
> > 
> > 
> > 为啥就变成了一个病句了。
> > 事务的基本保障是，我们需要分布式计算的异常处理机制。
> > 保障是我们需要xxx的机制
> > 或者难受就改变语序
> > 我们需要的机制是保障
> 
> “事务的保障是我们需要xxx的机制”事务的保障变成了我们需要。事务的保障应该是一个名词，而不是一个动词或者动作。

你这主语抽离出来了么。

保障是机制。 其他的都是修饰『保障』或者『机制』的。怎么就文字属性发生变更了？

Reply:
这一句我直接改回原翻译。

Reply:
> > > > > the fundamental premise of transactions is that we needed exception handling in distributed computations.
> > > > > 这个是我们需要分布式计算的异常处理机制。
> > > > > 不应该翻译成「我们所需的分布式计算中的异常处理」 赶紧很怪
> > > > 
> > > > 
> > > > 如果该成这样：“事务的基本保障是我们需要分布式计算的异常处理机制。”是个病句，事务的基本保障是我们的一种需要。
> > > 
> > > 
> > > 为啥就变成了一个病句了。
> > > 事务的基本保障是，我们需要分布式计算的异常处理机制。
> > > 保障是我们需要xxx的机制
> > > 或者难受就改变语序
> > > 我们需要的机制是保障
> > 
> > 
> > “事务的保障是我们需要xxx的机制”事务的保障变成了我们需要。事务的保障应该是一个名词，而不是一个动词或者动作。
> 
> 你这主语抽离出来了么。
> 
> 保障是机制。 其他的都是修饰『保障』或者『机制』的。怎么就文字属性发生变更了？

很简单，在里边加个“的”就没有语法问题了，改成“事务的基本保障是我们需要的分布式计算的异常处理机制。”但如果“需要”后没有“的”，确实是病句

Reply:
> > > > 如果程序的一个部分失败了
> > > > 这个 if 翻译成 「一旦」吧
> > > 
> > > 
> > > 如果和一旦相差不大，你觉得需要改就直接改。
> > 
> > 
> > 含义差距不大，但是从阅读的体验上面来看 「一旦」更好
> 
> 这本书错翻，暴力直译，不知所云的地方太多了。这点差别就是小巫见大巫。。。

都提 PR 修改了，能改就都改了。

Reply:
我这个pr允许被所有者编辑啊，你应该可以直接修改吧。你觉得有异议的地方，直接在上边修改就行了。我自己的仓库有自己修改版的，提pr只是个建议。